nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/model/impl/ListProperty.java

```diff
@@ -201,8 +198,7 @@ public Serializable getValueForWrite() throws PropertyException {
         if (children.isEmpty()) {
             return new ArrayList<String>();
         }
-        // noinspection CollectionDeclaredAsConcreteClass
-        ArrayList<Object> list = new ArrayList<>(children.size());
+        var list = new ArrayList<>(children.size());
```

Comment:
Since you are cleaning here, why not typing with <Property>  since it's used as such next line ?

Reply:
Because it's not needed, we never read this `list` again, we are only interested on its Serializable aspect.